test/vm_tests/transformer/declaration_parsing_test.dart

```diff
@@ -140,6 +140,45 @@ main() {
           expect(declarations.declaresComponent, isTrue);
         });
 
+        test('a component with a private \$ prefixed props class without a matching public class', () {
```

Comment:
We should have this test, but verify that the transformer throws an error when there is no accompanying public props class.